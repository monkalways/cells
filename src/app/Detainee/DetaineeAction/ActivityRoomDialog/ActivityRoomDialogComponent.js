import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';

import Loading from '../../../common/Loading';

const propTypes = {
  areActivityRoomsRefreshing: PropTypes.bool.isRequired,
  currentActivity: PropTypes.string.isRequired,
  detainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
  }).isRequired,
  getAvailableActivityRoomsRefresh: PropTypes.func.isRequired,
  isAnyRoomForGivenActivityAvailable: PropTypes.bool.isRequired,
  isDialogOpen: PropTypes.bool.isRequired,
  isAssigningToRoom: PropTypes.bool.isRequired,
  moveDetaineeToActivityRoom: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  destinationRoom: PropTypes.string,
  usage: PropTypes.string.isRequired,
};

const defaultProps = {
  destinationRoom: null,
};

const ActivityRoomDialogComponent = ({
  areActivityRoomsRefreshing,
  currentActivity,
  detainee,
  getAvailableActivityRoomsRefresh,
  isAnyRoomForGivenActivityAvailable,
  isAssigningToRoom,
  isDialogOpen,
  moveDetaineeToActivityRoom,
  onClose,
  destinationRoom,
  usage,
}) => {
  const handleClick = () => {
    moveDetaineeToActivityRoom(detainee.id, currentActivity, destinationRoom);
  };

  const handleOpen = () => {
    getAvailableActivityRoomsRefresh();
  };

  return (
    <Dialog
      disableBackdropClick={isAssigningToRoom}
      disableEscapeKeyDown={isAssigningToRoom}
      open={isDialogOpen}
      onClose={onClose}
      onRendered={handleOpen}
    >
      {/** * Medical, Fingerprinting, Telephone, Bail Hearing 1 and 2, Breath Test ** */}
      {areActivityRoomsRefreshing ? (
        <React.Fragment>
          <DialogTitle>{`Searching for available ${usage} room...`}</DialogTitle>
          <DialogContent>
            <Loading size={50} />
          </DialogContent>
        </React.Fragment>
      ) : isAnyRoomForGivenActivityAvailable ? (
        <React.Fragment>
          <DialogTitle>
            {`Moving ${detainee.lastName}, ${
              detainee.firstName
            } to ${usage} room ${destinationRoom}?`}
          </DialogTitle>
          {isAssigningToRoom && (
            <DialogContent>
              <Loading size={50} />
            </DialogContent>
          )}
          <DialogActions>
            <Button
              onClick={onClose}
              color="primary"
              disabled={isAssigningToRoom}
            >
              Cancel
            </Button>
            <Button
              onClick={handleClick}
              color="primary"
              variant="contained"
              autoFocus
              disabled={isAssigningToRoom}
            >
              Confirm
            </Button>
          </DialogActions>
        </React.Fragment>
      ) : (
        <React.Fragment>
          <DialogTitle>{`Sorry, the ${usage} room is no longer available.`}</DialogTitle>
          <DialogActions>
            <Button onClick={onClose} color="primary">
              Close
            </Button>
          </DialogActions>
        </React.Fragment>
      )}
    </Dialog>
  );
};

ActivityRoomDialogComponent.propTypes = propTypes;
ActivityRoomDialogComponent.defaultProps = defaultProps;

export default ActivityRoomDialogComponent;
