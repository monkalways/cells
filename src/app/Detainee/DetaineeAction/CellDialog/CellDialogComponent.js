import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Typography,
} from '@material-ui/core';

import Loading from '../../../common/Loading';
import constants from '../../constants';

const propTypes = {
  checkDetaineeInToCell: PropTypes.func.isRequired,
  currentActivity: PropTypes.string.isRequired,
  detainee: PropTypes.shape({
    assignedCellName: PropTypes.string.isRequired,
    currentActivityRoom: PropTypes.string,
    firstName: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
  }).isRequired,
  isDialogOpen: PropTypes.bool.isRequired,
  isAssigningToRoom: PropTypes.bool.isRequired,
  moveDetaineeToCell: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
};

const defaultProps = {};

const CellDialogComponent = ({
  checkDetaineeInToCell,
  currentActivity,
  detainee,
  isAssigningToRoom,
  isDialogOpen,
  moveDetaineeToCell,
  onClose,
}) => {
  const handleClick = () => {
    const { id, currentActivityRoom, assignedCellName } = detainee;
    if (currentActivity === constants.CELL) checkDetaineeInToCell(id, assignedCellName);
    else moveDetaineeToCell(id, currentActivityRoom, assignedCellName);
  };

  return (
    <Dialog
      disableBackdropClick={isAssigningToRoom}
      disableEscapeKeyDown={isAssigningToRoom}
      open={isDialogOpen}
      onClose={onClose}
    >
      <React.Fragment>
        {currentActivity === constants.CELL ? (
          <DialogTitle disableTypography>
            <Typography variant="h6" align="center">
              {`Placing ${detainee.lastName}, ${detainee.firstName} in Cell ${
                detainee.assignedCellName
              }.`}
            </Typography>
            <Typography variant="h6" align="center">
              Proceed with Action?
            </Typography>
          </DialogTitle>
        ) : (
          // <DialogTitle>
          //   {`Placing ${detainee.lastName}, ${detainee.firstName} in Cell ${
          //     detainee.assignedCellName
          //   }. Proceed with action?`}
          // </DialogTitle>
          <DialogTitle>
            {`Moving ${detainee.lastName}, ${detainee.firstName} to Cell ${
              detainee.assignedCellName
            }?`}
          </DialogTitle>
        )}
        {isAssigningToRoom && (
          <DialogContent>
            <Loading size={50} />
          </DialogContent>
        )}
        <DialogActions>
          <Button
            onClick={onClose}
            color="primary"
            disabled={isAssigningToRoom}
          >
            Cancel
          </Button>
          <Button
            onClick={handleClick}
            color="primary"
            variant="contained"
            autoFocus
            disabled={isAssigningToRoom}
          >
            Confirm
          </Button>
        </DialogActions>
      </React.Fragment>
    </Dialog>
  );
};

CellDialogComponent.propTypes = propTypes;
CellDialogComponent.defaultProps = defaultProps;

export default CellDialogComponent;
