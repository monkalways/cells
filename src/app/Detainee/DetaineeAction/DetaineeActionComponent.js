import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {
  Avatar,
  Card,
  CardHeader,
  CardContent,
  Grid,
  Typography,
  withStyles,
} from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';

import MedicalVisitIcon from '../../../images/MedicalVisit.png';
import PhoneAcceptIcon from '../../../images/PhoneAccept.png';
import PhoneDeclineIcon from '../../../images/PhoneDecline.png';
import FingerprintingIcon from '../../../images/Fingerprinting.png';
import BreathalyzerIcon from '../../../images/Breathalyzer.png';
import InterviewIcon from '../../../images/Interview.png';
import BailHearing1Icon from '../../../images/BailHearing1.png';
import BailHearing2Icon from '../../../images/BailHearing2.png';
import RemandHoldingIcon from '../../../images/RemandHolding.png';
import ReleaseHoldingIcon from '../../../images/ReleaseHolding.png';
import InCellIcon from '../../../images/InCell.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  isMedicalRoomOptionAvailable: PropTypes.bool.isRequired,
  isPhoneRoomOptionAvailable: PropTypes.bool.isRequired,
};

const DetaineeActionComponent = ({
  classes,
  isMedicalRoomOptionAvailable,
  isPhoneRoomOptionAvailable,
}) => (
  <Card className={classes.card}>
    <CardHeader
      component={() => (
        <Typography variant="h6" className={classes.cardHeader}>
          Detainee Action
        </Typography>
      )}
    />
    <CardContent className={classes.cardContent}>
      <Grid container spacing={16}>
        <Grid container item xs={12}>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton
              className={classes.button}
              disabled={!isMedicalRoomOptionAvailable}
            >
              <Avatar
                className={classNames(classes.avatar, {
                  [classes.disabled]: !isMedicalRoomOptionAvailable,
                })}
                src={MedicalVisitIcon}
              />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton
              className={classes.button}
              disabled={!isPhoneRoomOptionAvailable}
            >
              <Avatar
                className={classNames(classes.avatar, {
                  [classes.disabled]: !isPhoneRoomOptionAvailable,
                })}
                src={PhoneAcceptIcon}
              />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={PhoneDeclineIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={FingerprintingIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={BreathalyzerIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={InterviewIcon} />
            </IconButton>
          </Grid>
        </Grid>

        <Grid container item xs={12}>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={BailHearing1Icon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={BailHearing2Icon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={RemandHoldingIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={ReleaseHoldingIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={InCellIcon} />
            </IconButton>
          </Grid>
        </Grid>
      </Grid>
    </CardContent>
  </Card>
);
DetaineeActionComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  avatar: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
  button: {
    padding: theme.spacing.unit * 0.5,
  },
  cardContent: {
    backgroundColor: '#d4e2fc',
    paddingBottom: [16, '!important'],
  },
  cardHeader: {
    backgroundColor: '#3d89f7',
    paddingLeft: theme.spacing.unit,
    paddingTop: theme.spacing.unit * 0.5,
    paddingBottom: theme.spacing.unit * 0.5,
  },
  disabled: {
    opacity: '0.2',
  },
  imgContainer: {
    display: 'flex',
    justifyContent: 'center',
  },
}))(DetaineeActionComponent);
