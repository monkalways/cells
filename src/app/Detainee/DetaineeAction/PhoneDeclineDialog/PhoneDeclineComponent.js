import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';

import Loading from '../../../common/Loading';

const propTypes = {
  cellName: PropTypes.string,
  detainee: PropTypes.shape({
    arrestId: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  isDialogOpen: PropTypes.bool.isRequired,
  isUpdatingDetentionLog: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  savePhoneCallDecline: PropTypes.func.isRequired,
  userName: PropTypes.string,
};

const defaultProps = {
  cellName: null,
  userName: null,
};

const PhoneDeclineComponent = ({
  cellName,
  detainee,
  isDialogOpen,
  isUpdatingDetentionLog,
  onClose,
  savePhoneCallDecline,
  userName,
}) => {
  const handleClick = () => {
    const { arrestId } = detainee;
    savePhoneCallDecline(arrestId, cellName, userName);
  };

  return (
    <Dialog
      disableBackdropClick={isUpdatingDetentionLog}
      disableEscapeKeyDown={isUpdatingDetentionLog}
      open={isDialogOpen}
      onClose={onClose}
    >
      <React.Fragment>
        <DialogTitle>
          {`Record that ${detainee.lastName}, ${
            detainee.firstName
          } declined a phone call?`}
        </DialogTitle>
        {isUpdatingDetentionLog && (
          <DialogContent>
            <Loading size={50} />
          </DialogContent>
        )}
        <DialogActions>
          <Button
            id="cancelButton"
            onClick={onClose}
            color="primary"
            disabled={isUpdatingDetentionLog}
          >
            Cancel
          </Button>
          <Button
            id="confirmButton"
            onClick={handleClick}
            color="primary"
            variant="contained"
            autoFocus
            disabled={isUpdatingDetentionLog}
          >
            Confirm
          </Button>
        </DialogActions>
      </React.Fragment>
    </Dialog>
  );
};

PhoneDeclineComponent.propTypes = propTypes;
PhoneDeclineComponent.defaultProps = defaultProps;

export default PhoneDeclineComponent;
