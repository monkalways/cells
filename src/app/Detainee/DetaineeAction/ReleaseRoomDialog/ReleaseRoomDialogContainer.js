import { connect } from 'react-redux';
import { operations, selectors } from '../../duck';
import AutomaticRoomSelectionDialog from '../AutomaticRoomSelectionDialog';
import constants from '../../constants';

export const mapStateToProps = (
  state,
  areRoomsRefreshing = false, // *** fill this in
  currentRoom = selectors.getCurrentRoomState(state),
  destinationRoom = selectors.getFirstAvailableReleaseRoomState(state),
  isAnyRoomAvailable = selectors.isReleaseRoomOptionAvailableState(state),
  isAssigningToRoom = false, // *** fill this in
  usage = constants.RELEASE_ROOM,
) => ({
  areRoomsRefreshing,
  currentRoom,
  destinationRoom,
  isAnyRoomAvailable,
  isAssigningToRoom,
  usage,
});

export const mapDispatchToProps = (dispatch) => ({
  // eslint-disable-next-line max-len
  getAvailableRoomsRefresh: (getAvailableRoomsRefresh = operations.getAvailableReleaseRoomsRefresh) => {
    dispatch(getAvailableRoomsRefresh());
  },
  moveDetaineeToRoom: (
    detaineeId,
    originRoom,
    destinationRoom,
    goBack,
    moveDetaineeToActivityRoom = operations.moveDetaineeToRoom,
  ) => dispatch(moveDetaineeToActivityRoom(
    detaineeId,
    originRoom,
    destinationRoom,
    goBack,
  )),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AutomaticRoomSelectionDialog);
