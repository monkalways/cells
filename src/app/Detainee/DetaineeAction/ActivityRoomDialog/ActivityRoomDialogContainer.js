import { connect } from 'react-redux';
import { operations, selectors } from '../../duck';
import ActivityRoomDialogComponent from './ActivityRoomDialogComponent';

export const mapStateToProps = (
  state,
  { usage },
  areActivityRoomsRefreshing = selectors.areActivityRoomsRefreshingState(state),
  currentRoom = selectors.getCurrentRoomState(state),
  isAnyRoomForGivenActivityAvailable = selectors.isAnyRoomForGivenActivityAvailableState(
    state,
    usage,
  ),
  isAssigningToRoom = selectors.isAssigningToRoomState(state),
  destinationRoom = selectors.getFirstAvailableActivityRoom(state, usage),
) => ({
  areActivityRoomsRefreshing,
  currentRoom,
  isAnyRoomForGivenActivityAvailable,
  isAssigningToRoom,
  destinationRoom,
});

export const mapDispatchToProps = (dispatch) => ({
  // eslint-disable-next-line max-len
  getAvailableActivityRoomsRefresh: (getAvailableActivityRoomsRefresh = operations.getAvailableActivityRoomsRefresh) => dispatch(getAvailableActivityRoomsRefresh()),
  moveDetaineeToActivityRoom: (
    detaineeId,
    originRoom,
    destinationRoom,
    moveDetaineeToActivityRoom = operations.moveDetaineeToRoom,
  ) => dispatch(moveDetaineeToActivityRoom(detaineeId, originRoom, destinationRoom)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ActivityRoomDialogComponent);
