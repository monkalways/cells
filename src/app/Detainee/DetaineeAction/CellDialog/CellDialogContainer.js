import { connect } from 'react-redux';
import { operations, selectors } from '../../duck';
import CellDialogComponent from './CellDialogComponent';

export const mapStateToProps = (
  state,
  currentActivity = selectors.getCurrentActivityRoomState(state),
  isAssigningToRoom = selectors.isAssigningToRoomState(state),
) => ({
  currentActivity,
  isAssigningToRoom,
});

export const mapDispatchToProps = (dispatch) => ({
  moveDetaineeToCell: (
    detaineeId,
    from,
    destinationRoom,
    moveDetaineeToCell = operations.moveDetaineeToCell,
  ) => dispatch(moveDetaineeToCell(detaineeId, from, destinationRoom)),
  checkDetaineeInToCell: (detaineeId) => dispatch(operations.checkDetaineeInToCell(detaineeId)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CellDialogComponent);
