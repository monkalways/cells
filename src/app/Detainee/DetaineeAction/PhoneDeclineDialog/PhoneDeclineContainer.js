import { connect } from 'react-redux';
import { operations, selectors } from '../../duck';
import { selectors as authenticationSelectors } from '../../../Authentication/duck';
import PhoneDeclineComponent from './PhoneDeclineComponent';

export const mapStateToProps = (
  state,
  cellName = selectors.getCurrentRoomState(state),
  isUpdatingDetentionLog = selectors.isUpdatingDetentionLogState(state),
  userName = authenticationSelectors.getUserNameState(state),
) => ({
  cellName,
  isUpdatingDetentionLog,
  userName,
});

export const mapDispatchToProps = (dispatch) => ({
  savePhoneCallDecline: (
    arrestId,
    cellName,
    userName,
    savePhoneCallDecline = operations.savePhoneCallDecline,
  ) => {
    dispatch(savePhoneCallDecline(arrestId, cellName, userName));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PhoneDeclineComponent);
