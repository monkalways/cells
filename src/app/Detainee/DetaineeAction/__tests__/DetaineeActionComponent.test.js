import React from 'react';
import { shallow } from 'enzyme';

import constants from '../../constants';
import { DetaineeActionComponent } from '../DetaineeActionComponent';
import ActivityRoomDialog from '../ActivityRoomDialog';
import CellDialog from '../CellDialog';
import PhoneDeclineDialog from '../PhoneDeclineDialog';
import ReleaseRoomDialog from '../ReleaseRoomDialog'
import RemandRoomDialog from '../RemandRoomDialog'
import RoomSelectionDialog from '../RoomSelectionDialog';

describe('DetaineeActionComponent', () => {
  let detainee;
  let handleClose;
  let isBailHearingRoom1OptionAvailable;
  let isBailHearingRoom2OptionAvailable;
  let isBreathTestRoomOptionAvailable;
  let isFingerprintingRoomOptionAvailable;
  let isInCellOptionAvailable;
  let isInterviewRoomOptionAvailable;
  let isMedicalRoomOptionAvailable;
  let isPhoneDeclineOptionAvailable;
  let isPhoneRoomOptionAvailable;
  let isReleaseRoomOptionAvailable;
  let isRemandHoldingRoomOptionAvailable;

  beforeEach(() => {
    detainee = {
      firstName: 'Bob',
      id: '123',
      lastName: 'Saget',
    };
    handleClose = jest.fn();
    isBailHearingRoom1OptionAvailable = false;
    isBailHearingRoom2OptionAvailable = false;
    isBreathTestRoomOptionAvailable = false;
    isFingerprintingRoomOptionAvailable = false;
    isInCellOptionAvailable = false;
    isInterviewRoomOptionAvailable = false;
    isMedicalRoomOptionAvailable = false;
    isPhoneDeclineOptionAvailable = false;
    isPhoneRoomOptionAvailable = false;
    isReleaseRoomOptionAvailable = false;
    isRemandHoldingRoomOptionAvailable = false;
  });

  const setup = () => {
    const classes = {
      avatar: 'avatar',
      button: 'button',
      cardContent: 'cardContent',
      cardHeader: 'cardHeader',
      disabled: 'disabled',
      imgContainer: 'imgContainer',
    };
    return shallow(<DetaineeActionComponent
      classes={classes}
      detainee={detainee}
      handleClose={handleClose}
      isBailHearingRoom1OptionAvailable={isBailHearingRoom1OptionAvailable}
      isBailHearingRoom2OptionAvailable={isBailHearingRoom2OptionAvailable}
      isBreathTestRoomOptionAvailable={isBreathTestRoomOptionAvailable}
      isFingerprintingRoomOptionAvailable={
          isFingerprintingRoomOptionAvailable
        }
      isInCellOptionAvailable={isInCellOptionAvailable}
      isInterviewRoomOptionAvailable={isInterviewRoomOptionAvailable}
      isMedicalRoomOptionAvailable={isMedicalRoomOptionAvailable}
      isPhoneDeclineOptionAvailable={isPhoneDeclineOptionAvailable}
      isPhoneRoomOptionAvailable={isPhoneRoomOptionAvailable}
      isReleaseRoomOptionAvailable={isReleaseRoomOptionAvailable}
      isRemandHoldingRoomOptionAvailable={isRemandHoldingRoomOptionAvailable}
    />);
  };

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should set initial state', () => {
    const wrapper = setup();
    expect(wrapper.state().openDialog).toBeNull();
    expect(wrapper.state().usage).toEqual('');
  });

  it.each([
    ['#medical', constants.MEDICAL_ROOM],
    ['#phoneAccept', constants.PHONE_ROOM],
    ['#breathTest', constants.BREATH_TEST_ROOM],
    ['#bailHearing1', constants.BAIL_HEARING_ROOM_1],
    ['#bailHearing2', constants.BAIL_HEARING_ROOM_2],
  ])(
    'should update state when activity room button is clicked',
    (id, activityRoomUsage) => {
      const wrapper = setup();
      const activityRoomDialog = 'activityRoomDialog';

      wrapper.find(id).simulate('click');
      expect(wrapper.state().openDialog).toEqual(activityRoomDialog);
      expect(wrapper.state().usage).toEqual(activityRoomUsage);
    },
  );

  it('should update state when cell button is clicked', () => {
    const wrapper = setup();
    const cellDialog = 'cellDialog';

    wrapper.find('#cell').simulate('click');
    expect(wrapper.state().openDialog).toBe(cellDialog);
  });

  it('should update state when phone decline button is clicked', () => {
    const wrapper = setup();
    const phoneDeclineDialog = 'phoneDeclineDialog';

    wrapper.find('#phoneDecline').simulate('click');
    expect(wrapper.state().openDialog).toBe(phoneDeclineDialog);
  });

  it('should update state when release room button is clicked', () => {
    const wrapper = setup();
    const releaseRoomDialog = 'releaseRoomDialog';

    wrapper.find('#releaseHolding').simulate('click');
    expect(wrapper.state().openDialog).toBe(releaseRoomDialog);
  });

  it('should update state when remand room button is clicked', () => {
    const wrapper = setup();
    const remandRoomDialog = 'remandRoomDialog';

    wrapper.find('#remandHolding').simulate('click');
    expect(wrapper.state().openDialog).toBe(remandRoomDialog);
  });

  it.each([
    ['#fingerprinting', constants.FINGERPRINTING_ROOM],
    ['#interview', constants.INTERVIEW_ROOM],
  ])(
    'should update state when room selection button is clicked',
    (id, activityRoomUsage) => {
      const wrapper = setup();
      const roomSelectionDialog = 'roomSelectionDialog';

      wrapper.find(id).simulate('click');
      expect(wrapper.state().openDialog).toBe(roomSelectionDialog);
      expect(wrapper.state().usage).toEqual(activityRoomUsage);
    },
  );

  it.each([
    [ActivityRoomDialog],
    [CellDialog],
    [PhoneDeclineDialog],
    [ReleaseRoomDialog],
    [RemandRoomDialog],
    [RoomSelectionDialog],
  ])('should update state when modal is closed', (dialog) => {
    const wrapper = setup();

    wrapper.find(dialog).simulate('close');
    expect(wrapper.state().openDialog).toBeNull();
    expect(wrapper.state().usage).toBeNull();
    expect(handleClose).toBeCalledWith(detainee.id)
  });
});
