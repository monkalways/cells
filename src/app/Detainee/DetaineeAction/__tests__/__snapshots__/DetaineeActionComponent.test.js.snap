// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetaineeActionComponent should render 1`] = `
<WithStyles(Card)>
  <WithStyles(CardHeader)
    component={[Function]}
  />
  <WithStyles(CardContent)
    className="cardContent"
  >
    <WithStyles(Grid)
      container={true}
      spacing={16}
    >
      <WithStyles(Grid)
        container={true}
        item={true}
        xs={12}
      >
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="medical"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="MedicalVisit.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="phoneAccept"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="PhoneAccept.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="phoneDecline"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="PhoneDecline.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="fingerprinting"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="Fingerprinting.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="breathTest"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="Breathalyzer.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="interview"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="Interview.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        container={true}
        item={true}
        xs={12}
      >
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="bailHearing1"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="BailHearing1.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="bailHearing2"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="BailHearing2.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="remandHolding"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="RemandHolding.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="releaseHolding"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="ReleaseHolding.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          className="imgContainer"
          item={true}
          xs={2}
        >
          <WithStyles(IconButton)
            className="button"
            disabled={true}
            id="cell"
            onClick={[Function]}
          >
            <WithStyles(Avatar)
              className="avatar disabled"
              src="InCell.png"
            />
          </WithStyles(IconButton)>
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(CardContent)>
  <Connect(ActivityRoomDialogComponent)
    detainee={
      Object {
        "firstName": "Bob",
        "id": "123",
        "lastName": "Saget",
      }
    }
    isDialogOpen={false}
    onClose={[Function]}
    usage=""
  />
  <Connect(CellDialogComponent)
    detainee={
      Object {
        "firstName": "Bob",
        "id": "123",
        "lastName": "Saget",
      }
    }
    isDialogOpen={false}
    onClose={[Function]}
  />
  <Connect(PhoneDeclineComponent)
    detainee={
      Object {
        "firstName": "Bob",
        "id": "123",
        "lastName": "Saget",
      }
    }
    isDialogOpen={false}
    onClose={[Function]}
  />
  <Connect(WithStyles(RoomSelectionDialogComponent))
    detainee={
      Object {
        "firstName": "Bob",
        "id": "123",
        "lastName": "Saget",
      }
    }
    isDialogOpen={false}
    onClose={[Function]}
    usage=""
  />
</WithStyles(Card)>
`;
